### PR TITLE
WIP: wrapPython: only inject Python code, don't wrap with makeWrapper

### PR DESCRIPTION
**Pull request by catern**
###### Motivation for this change

This makes Python executables (such as meson) which look at sys.argv[0] to find the currently executing script, work fine without special-casing anything.

See https://github.com/NixOS/nixpkgs/issues/58130

Still to resolve:

- We're unconditionally adding -s to the Python shebang to disable user site reading; we can either merge that option with others, or fall back to using a shell script if there are already options in the shebang.
- I removed makeWrapper entirely to demonstrate the potential, but we should probably still have makeWrapper be used if makeWrapperArgs is passed

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by FRidh** [comment]
This is essentially a continuation of https://github.com/NixOS/nixpkgs/pull/17749.

**Comment by FRidh** [comment]
About PYTHONNOUSERSITE, https://github.com/NixOS/nixpkgs/pull/51641. I suggest making `-s` optional. In case `makeWrapperArgs` is not empty, create a wrapper, otherwise, don't. Maybe in time we can deprecate it. With these changes done we can create a job and test it.

File: pkgs/development/interpreters/python/wrap-python.nix [hunk 1]
```diff
@@ -29,14 +29,14 @@ makeSetupHook {
         '';
 
         # This preamble does two things:
-        # * Sets argv[0] to the original application's name; otherwise it would be .foo-wrapped.
-        #   Python doesn't support `exec -a`.
+        # * Adds all required binary directories to PATH
```
Comment (FRidh) [inline pkgs/development/interpreters/python/wrap-python.nix]: Append, prepend ?

**Comment by siraben** [comment]
Still important to me.

**Comment by catern** [comment]
Alas, I haven't done any further work on this, and at this point I probably won't, so I'll go ahead and close this. But I would love to see something like this merged - I still have code which would benefit from this.

**Comment by mkenigs** [comment]
What's still left to be done on what you had written?

**Comment by catern** [comment]
IIRC I thought the change was essentially good in its current state (although of course it would be changed some in review). So if someone wanted to resurrect the change, I think it would just be a matter of rebasing, updating for @FRidh 's comments and opening a new PR. Also you probably would want to target the change at staging, and do some testing of packages that are affected by this which are mentioned in the linked issues (like qutebrowser).

**Comment by catern** [comment]
I'm not sure - there might not be any other argument passed to the shebang in current Nixpkgs.

**Pull request closed without merging**